addons/l10n_sa/models/account_chart_template.py

```diff
@@ -0,0 +1,62 @@
+                                                                                  code_digits=code_digits,
+                                                                                  account_ref=account_ref,
+                                                                                  taxes_ref=taxes_ref)
+        ifrs_journal_id = self.env['account.journal'].search([('company_id', '=', company.id), ('code', '=', 'IFRS')], limit=1)
+        if ifrs_journal_id:
+            ifrs_account_ids = [self.env.ref('l10n_sa.sa_account_100101').id,
+                                self.env.ref('l10n_sa.sa_account_100102').id,
+                                self.env.ref('l10n_sa.sa_account_400070').id]
+            ifrs_accounts = self.env['account.account'].browse([account_ref.get(id) for id in ifrs_account_ids])
+            for account in ifrs_accounts:
+                account.allowed_journal_ids = [(4, ifrs_journal_id.id, 0)]
+        zakat_journal_id = self.env['account.journal'].search([('company_id', '=', company.id), ('code', '=', 'ZAKAT')], limit=1)
+        if zakat_journal_id:
+            zakat_account_ids = [self.env.ref('l10n_sa.sa_account_201019').id,
+                                self.env.ref('l10n_sa.sa_account_400072').id]
+            zakat_accounts = self.env['account.account'].browse([account_ref.get(id) for id in zakat_account_ids])
+            for account in zakat_accounts:
+                account.allowed_journal_ids = [(4, zakat_journal_id.id, 0)]
+        self.env.ref('l10n_sa.sa_tax_group_taxes_15').write(
+            {'property_tax_payable_account_id': account_ref.get(self.env.ref('l10n_sa.sa_account_202003').id),
+             'property_tax_receivable_account_id': account_ref.get(self.env.ref('l10n_sa.sa_account_100103').id)})
+        self.env.ref('l10n_sa.sa_tax_group_taxes_withholding').write(
+            {'property_tax_payable_account_id': account_ref.get(self.env.ref('l10n_sa.sa_account_201020').id),
+             'property_tax_receivable_account_id': account_ref.get(self.env.ref('l10n_sa.sa_account_100103').id)}) # TODO Get correct account from Nada
+        return account_ref, taxes_ref
+
+    def _load(self, sale_tax_rate, purchase_tax_rate, company):
+        res = super(AccountChartTemplate, self)._load(sale_tax_rate, purchase_tax_rate, company)
+        company.default_cash_difference_expense_account_id.with_context(lang='ar_001').name = 'خسارة الفرق النقدي'
+        company.default_cash_difference_income_account_id.with_context(lang='ar_001').name = 'مكاسب الفرق النقدي'
+        return res
+
+    @api.model
+    def _create_liquidity_journal_suspense_account(self, company, code_digits):
+        account = super(AccountChartTemplate, self)._create_liquidity_journal_suspense_account(company=company,
+                                                                                               code_digits=code_digits)
+        account.with_context(lang='ar_001').name = 'حساب البنك المعلق'
```

Comment:
What's this ?

Reply:
I see the point in putting an Arab name for the automatically generated cash accounts, ...  Just wondering if it is not overkill/prone to errors.